src/wp-includes/class-wp-query.php

```diff
@@ -3521,8 +3521,20 @@ public function get_posts() {
 			/** @var WP_Post[] */
 			$this->posts = array_map( 'get_post', $this->posts );
 
-			if ( $q['cache_results'] ) {
-				update_post_caches( $this->posts, $post_type, $q['update_post_term_cache'], $q['update_post_meta_cache'] );
+			if ( $q['cache_results'] && $id_query_is_cacheable ) {
```

Comment:
I don't think we should use the `$id_query_is_cacheable` variable here because only part of that is relevant: All existing usage of this variable is around caching post query results (in `post-queries` group), but here we're only caching individual post data, so that has different implications. For example, the presence of `RAND(` in the orderby query does not matter for individual post caches.

Maybe we introduce a new variable `$id_posts_are_cacheabl` for the `$cacheable_field_values` logic, and then we can rely on that here.